### PR TITLE
Jusi 760 release tagging

### PR DESCRIPTION
**Pull request by super-cob**

### CODE REVIEW

File: tools/image-publish [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    branch_name = subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"]).decode("utf-8").strip()
+    assert branch_name == "main", "cannot publish and tag non-main git branch"
+
+    # Confirm that local and remote are at the same revision
+    subprocess.check_call(["git", "fetch"])
+    our_hash = subprocess.check_output(["git", "rev-parse", "HEAD"])
+    upstream_hash = subprocess.check_output(["git", "rev-parse", "@{u}"])
+    
+    # assert our_hash == upstream_hash, "local and/or remote are missing commits"
```
Comment (cslocum) [inline tools/image-publish:27]: Why is this commented out?  Seems important unless I'm missing something.

File: tools/image-publish [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    subprocess.run(["image-promote", args.registry, args.repo, args.old_tag, args.tag], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
+    subprocess.check_call(["git", "tag", "-a", args.tag, "-m", "auto tag version"])
+    subprocess.check_call(["git", "push", "--tags"])
```
Comment (cslocum) [inline tools/image-publish:30]: Some error handling here would probably be a good idea, since failures won't tank this script's execution.

**Comment by cslocum** [comment]
This looks super.  Can't merge without associated documentation updates though.

File: tools/image-publish (hunk 2 above)
Comment (super-cob) [reply tools/image-publish]: a check=True or a check_call will raise a CalledProcessException and tank the execution, i did update this to make sure to report the stderr to give some debugging help

Comment (cslocum) [reply tools/image-publish]: Super, thanks.

**Comment by cslocum** [comment]
Any pushed images should have `unscanned-` prefixes.

**Comment by super-cob** [comment]
> Any pushed images should have `unscanned-` prefixes.

I'm not pushing images though just tags

**Pull request merged**